services/projects-service/src/services/imageService.ts

```diff
@@ -158,3 +163,70 @@ export const updateXnatUserProfile = async (
         logger.error("No trusts reported back with the status. An empty list will be returned");
     }
 };
+
+export const reimportFailedStudies = async (
+    reimportQueries: IReimportQuery[],
+    connection: Sequelize,
+    projectReimportRate: number
+): Promise<boolean> => {
+    const reimportResponses: IReimportResponse[] = [];
+
+    await Promise.all(
+        reimportQueries.map(async (query) => {
+            const encodedQuery = Base64Url.encode(query.query);
+            const url = `${query.trustEndpoint}/imaging/${query.projectId}?query=${encodedQuery}`;
+
+            const PRIVATE_API_KEY = process.env.PRIVATE_API_KEY;
+        
+            if (!PRIVATE_API_KEY) {
+                logger.object({ message: "PRIVATE_API_KEY is not defined." });
+
+                throw new Error();
+            }
+
+            const lastReimportTime = query.lastReimport?.getTime() ?? 0;
+
+            if (Date.now() > lastReimportTime + projectReimportRate * (60 * 1000)) {
+                return httpRequest.put<void>(url, undefined, undefined, PRIVATE_API_KEY)
+                    .then(async (response) => {
+                        reimportResponses.push({
+                            projectId: query.projectId,
+                            trustId: query.trustId,
+                            trustName: query.trustName,
+                            status: response.status
+                        });
+
+                        if (response.status === HttpStatusCode.OK) {
+                            await connection.query(
+                                `UPDATE xnatprojectstatus 
+                                    SET lastreimport = now(),
+                                        reimportcount = reimportcount +1
+                                    WHERE xnatprojectstatus.projectid=:projectId
+                                    AND xnatprojectstatus.trustid=:trustId`,
+                                {
+                                    type: QueryTypes.UPDATE,
+                                    replacements: { 
+                                        projectId: query.projectId,
+                                        trustId: query.trustId
+                                    }
+                                }
+                            );
+                        }
+                    }).catch((e) => {
+                        logger.object({
+                            message: `There was an error when reimporting studies for project ${query.projectId} at trust ${query.trustName}`,
+                            error: e.response?.data ?? e.message
+                        });
+                    });  
+            } else {
+                logger.object({ message: `Time specified in PROJECT_REIMPORT_RATE has not passed for project ${query.projectId} at trust ${query.trustName}` });
```

Comment:
Very nit picky but could change this to
```suggestion
                logger.object({ message: `Time specified in PROJECT_REIMPORT_RATE has not exceeded for project ${query.projectId} at trust ${query.trustName}` });
```

Reply:
Yeah cool. Would it be 'has not been exceeded'?

Reply:
yeah that also works